uc-crux-llvm/src/UCCrux/LLVM/Run/Check.hs

```diff
@@ -0,0 +1,406 @@
+    SomeCheckResult(..),
+    SomeCheckedCalls,
+    getSomeCheckedCalls,
+    checkResultFunContext,
+    checkResult,
+    checkInferredContracts,
+    checkInferredContractsFromEntryPoints,
+    inferThenCheck,
+    ppSomeCheckResult,
+  )
+where
+
+{- ORMOLU_DISABLE -}
+import           Control.Lens ((^.), to)
+import           Data.Foldable (toList)
+import qualified Data.IORef as IORef
+import           Data.Map.Strict (Map)
+import qualified Data.Map.Strict as Map
+import           Data.Maybe (catMaybes)
+import qualified Data.Text as Text
+import           Data.Traversable (for)
+import           Data.Type.Equality ((:~:)(Refl), testEquality)
+
+import qualified Prettyprinter as PP
+
+import           Data.Parameterized.Context (Assignment, Ctx)
+import qualified Data.Parameterized.Context as Ctx
+import           Data.Parameterized.Classes (IxedF'(ixF'))
+import           Data.Parameterized.Some (Some(Some))
+
+import qualified What4.Interface as What4
+
+-- crucible
+import           Lang.Crucible.Backend (IsSymInterface)
+import qualified Lang.Crucible.CFG.Core as Crucible
+import qualified Lang.Crucible.Simulator.RegMap as Crucible
+import           Lang.Crucible.FunctionHandle (HandleAllocator)
+
+-- crucible-llvm
+import           Lang.Crucible.LLVM.MemModel (MemImpl, HasLLVMAnn)
+import           Lang.Crucible.LLVM.Extension (LLVM)
+
+-- crux
+import           Crux.Config.Common (CruxOptions)
+import qualified Crux.Log as Crux
+import qualified Crux.Types as Crux
+
+-- crux-llvm
+import           Crux.LLVM.Config (LLVMOptions)
+import qualified Crux.LLVM.Config as CruxLLVM
+import           Crux.LLVM.Overrides (ArchOk)
+
+-- local
+import           UCCrux.LLVM.Constraints (Constraints, emptyConstraints, ppConstraint, ppShapeConstraint)
+import           UCCrux.LLVM.Cursor (ppSelector)
+import           UCCrux.LLVM.Context.App (AppContext)
+import           UCCrux.LLVM.Context.Module (ModuleContext, CFGWithTypes(..), findFun)
+import           UCCrux.LLVM.Context.Function (FunctionContext, makeFunctionContext, ppFunctionContextError, argumentFullTypes)
+import           UCCrux.LLVM.Errors.Panic (panic)
+import           UCCrux.LLVM.FullType (FullType, FullTypeRepr, MapToCrucibleType)
+import qualified UCCrux.LLVM.FullType.CrucibleType as FT
+import           UCCrux.LLVM.Module (DefnSymbol, FuncSymbol(FuncDefnSymbol), defnSymbolToString)
+import qualified UCCrux.LLVM.Overrides.Check as Check
+import qualified UCCrux.LLVM.Overrides.Stack as Stack
+import           UCCrux.LLVM.PP (ppRegMap)
+import           UCCrux.LLVM.Run.EntryPoints (EntryPoints, getEntryPoints)
+import qualified UCCrux.LLVM.Run.Simulate as Sim
+import qualified UCCrux.LLVM.Run.Loop as Loop
+import           UCCrux.LLVM.Run.Result (SomeBugfindingResult)
+import qualified UCCrux.LLVM.Run.Result as Result
+import           UCCrux.LLVM.Setup (SymValue)
+import           UCCrux.LLVM.Shape (Shape)
+{- ORMOLU_ENABLE -}
+
+newtype SomeCheckedCalls m sym arch =
```

Comment:
This type looks complicated enough to warrant an explanation. What is it for? What should a caller think of that inner continuation as?

Reply:
You're absolutely right thanks. Added comments.